feat(gossipsub): add getter function to obtain `TopicScoreParams`

ackintosh opened the pull request:
## Description

<!--
Please write a summary of your changes and why you made them.
This section will appear as the commit message after merging.
Please craft it accordingly.
For a quick primer on good commit mesages, check out this blog post: https://cbea.ms/git-commit/

Please include any relevant issues in here, for example:

Related https://github.com/libp2p/rust-libp2p/issues/ABCD.
Fixes https://github.com/libp2p/rust-libp2p/issues/XYZ.
-->

Added a getter function to obtain `TopicScoreParams`. I need this function for testing which checks if scoring parameters are updated as expected.

## Notes & open questions

<!--
Any notes, remarks or open questions you have to make about the PR which don't need to go into the final commit message.
-->

## Change checklist

<!-- Please add a Changelog entry in the appropriate crates and bump the crate versions if needed. See <https://github.com/libp2p/rust-libp2p/blob/master/docs/release.md#development-between-releases>-->

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
  - I haven't added a test as the function added is just getter but let me know if you think it is needed.
- [x] A changelog entry has been made in the appropriate crates

ackintosh committed:
Add get_topic_params

ackintosh committed:
Add changelog entry

ackintosh committed:
Make clippy happy

divagant-martian commented on protocols/gossipsub/src/peer_score.rs:
```diff
@@ -780,6 +780,11 @@ impl PeerScore {
     }
 
+    /// Returns a scoring parameters for a topic if existent.
+    pub(crate) fn get_topic_params(&self, topic_hash: &TopicHash) -> Option<TopicScoreParams> {
```
should be enough to return a reference. If the caller needs a value they can clone it

divagant-martian commented on protocols/gossipsub/src/behaviour.rs:
```diff
@@ -930,6 +930,15 @@ where
     }
 
+    /// Returns a scoring parameters for a topic if existent.
+    pub fn get_topic_params<H: Hasher>(&self, topic: &Topic<H>) -> Option<TopicScoreParams> {
```
should be enough to return a reference. If the caller needs a value they can clone it

divagant-martian commented on protocols/gossipsub/CHANGELOG.md:
```diff
@@ -1,4 +1,10 @@
-## 0.45.0 
+## 0.45.1 - unreleased
```
recently (today) I learned you need to change the `Cargo.toml` upgrading the version

ackintosh committed:
Return a reference to avoid needless clones

ackintosh committed:
Update Cargo.toml

thomaseizinger commented on protocols/gossipsub/src/behaviour.rs:
```diff
@@ -930,6 +930,15 @@ where
+            peer_score.get_topic_params(&topic.hash())
+        } else {
+            None
+        }
```
This can be expressed using `?` to bubble up the `None` case :)

thomaseizinger commented on protocols/gossipsub/CHANGELOG.md:
```diff
@@ -1,4 +1,10 @@
+
+- Add getter function to obtain `TopicScoreParams`.
+
+[PR 4231]: https://github.com/libp2p/rust-libp2p/pull/4231
```
The link reference does nothing if it is not used!

```suggestion
- Add getter function to obtain `TopicScoreParams`.
  See [PR 4231].

[PR 4231]: https://github.com/libp2p/rust-libp2p/pull/4231
```

thomaseizinger commented on protocols/gossipsub/Cargo.toml:
```diff
@@ -3,7 +3,7 @@ name = "libp2p-gossipsub"
 rust-version = { workspace = true }
 description = "Gossipsub protocol for libp2p"
-version = "0.45.0"
+version = "0.45.1"
```
CI complains that the lockfile is not up to date.

thomaseizinger replied:
CI complains that the lockfile is not up to date.

thomaseizinger reviewed (approved):
Thanks!

I think exposing this makes sense. I left some minor comments :)

ackintosh committed:
Update protocols/gossipsub/CHANGELOG.md

Co-authored-by: Thomas Eizinger <thomas@eizinger.io>

ackintosh committed:
Use `?` to bubble up the `None` case

ackintosh committed:
Update Cargo.lock

ackintosh committed:
Fix compile error

error[E0507]: cannot move out of `self.peer_score` which is behind a shared reference

thomaseizinger reviewed (dismissed):
LGTM, thanks!

mxinden committed:
Merge branch 'master' into feature/getter-for-topic-params
